src/main/java/xyz/nucleoid/extras/statistics/NucleoidStatistics.java

```diff
@@ -0,0 +1,78 @@
+package xyz.nucleoid.extras.statistics;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import net.minecraft.util.Util;
+import org.apache.commons.io.IOUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import xyz.nucleoid.extras.NucleoidExtrasConfig;
+import xyz.nucleoid.plasmid.event.GameEvents;
+import xyz.nucleoid.plasmid.game.stats.GameStatisticBundle;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+
+public class NucleoidStatistics {
+    private static final Logger LOGGER = LogManager.getLogger();
+    private static final Gson GSON = new Gson();
+
+    private static NucleoidStatistics instance;
+    private static boolean initialized;
+
+    private final StatisticsBackendConfig config;
+
+    private NucleoidStatistics(StatisticsBackendConfig config) {
+        this.config = config;
+        this.attachEventListener();
+    }
+
+    private void attachEventListener() {
+        GameEvents.CLOSING.register((space, reason) -> {
+            // TODO: Should we only submit statistics on reason == FINISHED?
+            for (Map.Entry<String, GameStatisticBundle> entry : space.getAllStatistics().entrySet()) {
+                this.submitStatisticBundle(entry.getKey(), entry.getValue());
+            }
+        });
+    }
+
+    private void submitStatisticBundle(String namespace, GameStatisticBundle bundle) {
+        Util.getIoWorkerExecutor().execute(() -> {
+            LOGGER.debug("Submitting statistic bundle for '{}'...", namespace);
+
+            JsonObject body = new JsonObject();
+            body.addProperty("server_name", this.config.getServerName());
+            body.addProperty("namespace", namespace);
+            body.add("stats", bundle.encodeBundle());
+
+            try {
+                URL url = new URL(this.config.getBackendUrl() + "/stats/upload");
```

Comment:
Hmm, should this be a separate HTTP endpoint or does including it within the existing backend mechanism make the most sense?

Reply:
Currently its a [separate backend service](https://github.com/Tom-The-Geek/nucleoid-persistence-backend), that isn't part of the normal backend, and so it just uses a HTTP request to send the bundle. I can work on integrating it into the current backend at some point, but for now it just uses a simple HTTP API.

Reply:
Now uses the integrations connection to upload statistics